### PR TITLE
Recalculate bit when a config object changes

### LINKED ISSUE #3192: Question OOIs are not properly handled in backend
**Describe the bug**
Data that is impacted by the Question OOIs are not properly handled by the backend.

The logs do not contain any error messages relating to this bug. 

**To Reproduce**

1. Go through the onboarding flow.
2. Add HTTPHeader object for mispo.es with as key: `Content-Security-Policy` and as value `default-src https://githubpages.com`.
3. Enable Nmap TCP + Webpage analysis and give mispo.es Hostname L2 clearance.
4. Go to the `Question` object for `disallowed-csp-hostnames`. Click the submit button. 
6. Wait until all tasks are done. 
7. Check on the object page the page for the CSP header with a reference to the github pages. 
8. Observe that no findings are created for this object.  

**Expected behavior**
Output from the Question object should be parsed into findings. 

**Screenshots**
image

image

**OpenKAT version**
main 

**Additional context**
This bug was identified while doing the QA for #3165. This PR hasn't been properly checked and should thus be picked up at a later point in time.

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes

Recalculate bit when a config object changes, to address "Question OOIs are not properly handled in backend" as described #3192.

### Issue link

Closes #3192.

### Demo

image

image

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/octopoes/core/service.py [hunk 1]
```diff
@@ -400,14 +402,25 @@ def _on_update_ooi(self, event: OOIDBEvent) -> None:
...
-        inference_params = self.origin_parameter_repository.list_by_reference(
-            event.new_data.reference, valid_time=event.valid_time
-        )
-        for inference_param in inference_params:
-            inference_origins.append(self.origin_repository.get(inference_param.origin_id, event.valid_time))
+        if isinstance(event.new_data, Config):
+            relevant_bits = [bit for bit in get_bit_definitions().values() if bit.config_ooi_relation_path is not None]
+            inference_origins = list(
+                {
+                    x.model_dump_json(): x
```
[the remark this answers is not in the document]
Comment (originalsouth) [reply octopoes/octopoes/core/service.py:409]: Ideally I would use a set here, but `x` is unhashable; so this is a safe workaround. A set would do the same thing, but with the hash.

File: octopoes/octopoes/core/service.py [hunk 2]
```diff
@@ -400,14 +402,25 @@ def _on_update_ooi(self, event: OOIDBEvent) -> None:
...
+                    for x in chain.from_iterable(
+                        map(lambda x: self.origin_repository.list_origins(event.valid_time, method=x.id), relevant_bits)
```
[the remark this answers is not in the document]
Comment (originalsouth) [reply octopoes/octopoes/core/service.py]: Good suggestion! See 97b07513ad3f4046a4b9dfd2de97ca67e8e586c2.

**Review by ammar92 — APPROVED** [review]
Nice job! No further remarks

**Comment by originalsouth** [comment]
Merge main: angry Sigrid 
`¯\_(ツ)_/¯`

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Very nice work! Answering the Question results in the Config and thus the corresponding CSP finding. Updating the Question removes the finding automatically. I also tested if re-adding the bad hostname would would and it did. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Pull request merged**